apps/api/src/app/notifications/usecases/get-activity-feed/get-activity-feed.usecase.ts

```diff
@@ -1,23 +1,44 @@
-import { Injectable } from '@nestjs/common';
-import { NotificationFeedItemEntity, NotificationRepository, SubscriberRepository } from '@novu/dal';
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import {
+  CommunityOrganizationRepository,
+  NotificationFeedItemEntity,
+  NotificationRepository,
+  OrganizationEntity,
+  SubscriberRepository,
+} from '@novu/dal';
 import { Instrument } from '@novu/application-generic';
+import { ApiServiceLevelEnum, FeatureNameEnum, getFeatureForTierAsNumber } from '@novu/shared';
 import { ActivitiesResponseDto, ActivityNotificationResponseDto } from '../../dtos/activities-response.dto';
 import { GetActivityFeedCommand } from './get-activity-feed.command';
 import { mapFeedItemToDto } from './map-feed-item-to.dto';
 
+const DEFAULT_AFTER_DATE = new Date(Date.now() - 24 * 60 * 60 * 1000); // 24 hours ago
+
 @Injectable()
 export class GetActivityFeed {
   constructor(
     private subscribersRepository: SubscriberRepository,
-    private notificationRepository: NotificationRepository
+    private notificationRepository: NotificationRepository,
+    private organizationRepository: CommunityOrganizationRepository
   ) {}
 
+  /* eslint-disable no-param-reassign */
   async execute(command: GetActivityFeedCommand): Promise<ActivitiesResponseDto> {
     let subscriberIds: string[] | undefined;
 
+    const { after, before } = await this.validateRetentionLimitForTier(
+      command.organizationId,
+      command.after,
+      command.before
+    );
```

Comment:
Changed to always set `$gte $lte` boundaries on queries, its safer and easier to trace in case of issues